packages/plugin-ext/src/plugin/plugin-manager.ts

```diff
@@ -291,6 +291,14 @@ export class PluginManagerExtImpl implements PluginManagerExt, PluginManager {
                     // see https://github.com/TypeFox/vscode/blob/70b8db24a37fafc77247de7f7cb5bb0195120ed0/src/vs/workbench/api/common/extHostExtensionService.ts#L372-L376
                     pluginMain = pluginMain || {};
                     return await this.startPlugin(plugin, configStorage, pluginMain);
+                } catch (err) {
+                    if (this.pluginActivationPromises.has(plugin.model.id)) {
```

Comment:
Could we somehow get rid of duplicate code? maybe `startPlugin` should return plugin export but the rest be handled here?

Reply:
There is a lot of duplicated code in this file. Recommend de-duplication is revisited in a separate PR.

Reply:
ok

Reply:
It wasn't hard to fix :) Updated